src/sentry/sentry_apps/apps.py

```diff
@@ -143,7 +144,18 @@ def _update_scopes(self) -> None:
                 and self.sentry_app.scope_list != self.scopes
             ):
                 raise APIError("Cannot update permissions on a published integration.")
+
+            # We are using a pre_save signal to enforce scope hierarchy on the ApiToken model.
+            # Because we're using bulk_update here to update all the tokens for the SentryApp,
+            # we need to manually enforce the hierarchy because the pre_save signal won't be called.
+            new_scopes = set(self.scopes)
+            for scope in self.scopes:
+                if scope in SENTRY_SCOPES:
+                    new_scopes = new_scopes.union(SENTRY_SCOPE_HIERARCHY_MAPPING[scope])
+            self.scopes = sorted(new_scopes)
```

Comment:
is it possible to unify this with the receiver in a util function because the logic is the same?

Reply:
Good suggestion, created `add_scope_hierarchy` helper func